solana-programs/staking-bridge/tests/staking-bridge.ts

```diff
@@ -1,16 +1,293 @@
-import * as anchor from "@coral-xyz/anchor";
-import { Program } from "@coral-xyz/anchor";
-import { StakingBridge } from "../target/types/staking_bridge";
+import * as anchor from '@coral-xyz/anchor'
+import { Program } from '@coral-xyz/anchor'
+import { StakingBridge } from '../target/types/staking_bridge'
+import { CHAIN_ID_ETH } from '@certusone/wormhole-sdk'
+import {
+  TOKEN_PROGRAM_ID,
+  getOrCreateAssociatedTokenAccount,
+} from '@solana/spl-token'
+import {
+  ETH_AUDIO_TOKEN_ADDRESS,
+  SOL_AUDIO_DECIMALS,
+  SOL_AUDIO_TOKEN_ADDRESS,
+  SOL_USDC_TOKEN_ADDRESS,
+  ammProgram,
+  bridgeId,
+  serumDexProgram,
+  serumMarketPublicKey,
+  tokenBridgeId
+} from './constants'
+import { getAssociatedPoolKeys, getMarket, getVaultOwnerAndNonce } from './raydiumTestUtils'
+import { formatEthAddress, getMinimumAmountOutFromAmountIn, getPostMessageData } from './wormholeTestUtils'
 
-describe("staking-bridge", () => {
-  // Configure the client to use the local cluster.
-  anchor.setProvider(anchor.AnchorProvider.env());
+const {
+  Connection,
+  Keypair,
+  PublicKey,
+  SystemProgram,
+  SYSVAR_CLOCK_PUBKEY,
+  SYSVAR_RENT_PUBKEY,
+} = anchor.web3
+
+// Read in fee payer from the environment variable
+const FEE_PAYER_SECRET = process.env.feePayerSecret
+const feePayerKeypair = Keypair.fromSecretKey(
+  Uint8Array.from(JSON.parse(FEE_PAYER_SECRET))
+)
+const feePayerPublicKey = feePayerKeypair.publicKey
+
+const SOL_USDC_TOKEN_ADDRESS_KEY = new PublicKey(SOL_USDC_TOKEN_ADDRESS)
+const SOL_AUDIO_TOKEN_ADDRESS_KEY = new PublicKey(SOL_AUDIO_TOKEN_ADDRESS)
+
+const endpoint = 'https://api.mainnet-beta.solana.com'
+const connection = new Connection(endpoint, 'confirmed');
+
+async function signTransaction(transaction) {
+  transaction.partialSign(feePayerKeypair)
+  return transaction
+}
+async function signAllTransactions(transactions: any[]) {
+  transactions.forEach((transaction) => {
+    transaction.partialSign(feePayerKeypair)
+  })
+  return transactions
+}
+
+describe('staking-bridge', () => {
+  const wallet = {
+    publicKey: feePayerPublicKey,
+    signTransaction,
+    signAllTransactions
+  }
+  const provider = new anchor.AnchorProvider(connection, wallet, { skipPreflight: true });
+  anchor.setProvider(provider)
 
   const program = anchor.workspace.StakingBridge as Program<StakingBridge>;
 
-  it("Is initialized!", async () => {
+  const [stakingBridgePda, stakingBridgePdaBump] = PublicKey.findProgramAddressSync(
+    [Buffer.from('staking_bridge')],
+    program.programId
+  )
+
+  it('creates the staking bridge pda', async () => {
+    // Add your test here.
+    const tx = await program.methods
+      .createPda()
+      .accounts({
+        stakingBridgePda,
+        payer: feePayerPublicKey,
+        systemProgram: SystemProgram.programId,
+      })
+      .rpc();
+    console.log('Your transaction signature', tx);
```

Comment:
should there be an assert here?

Reply:
yeah missed the asserts. will handle here and in other places